drivers/wifi/siwx917/Kconfig.siwx917

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2023 Antmicro
+# SPDX-License-Identifier: Apache-2.0
+
+config WIFI_SIWX917
+	bool "Silabs SiWx917 SoC series WiFi driver"
+	default y
+	depends on DT_HAS_SILABS_SIWX917_WIFI_ENABLED
+	# FIXME: Explain why
+	depends on SYS_CLOCK_TICKS_PER_SEC <= 1000
+	select WIFI_OFFLOAD
+	select NET_L2_WIFI_MGMT
+	select NET_IPV4
+	select CMSIS_RTOS_V2
+	# Also select CMSIS-RTOS v2 dependencies
+	select POLL
+	select DYNAMIC_THREAD
+	select THREAD_NAME
+	select THREAD_STACK_INFO
+	select THREAD_MONITOR
+	select INIT_STACKS
```

Comment:
These don't seem like features that should be explicitly and unconditionally selected by a driver. I'm quite sure the driver can build without them (and if it can't this should be fixed in the driver).

Reply:
Btw, in general, `depends on` is preferred over `select`, although there does exist situations where `select` is the right choice too.

Reply:
These lines are results of `select CMSIS_RTOS_V2`. Let's say I use `depends on CMSIS_RTOS_V2`, 'samples/net/wifi/' does not enable `CMSIS_RTOS_V2`, so WiFi won't be enabled. How to deal with that?

Reply:
Ah. I missed the comment you have above these. They seem like odd mandatory dependencies for CMSIS-RTOS, but if that's how it is then so be it. So feel free to ignore my comment :)